### PR TITLE
468 log with emphasis wraps filenames poorly

### PR DESCRIPTION
**Pull request by biosafetylvl5**
* [x] NO REQUIRED **existing tests** (explain why not required)
* [x] Required **unit tests** added and pass for new/modified functionality
* [x] NO REQUIRED **integration tests** (explain why not required)
* [x] NO REQUIRED **documentation** (explain why not required)
* [x] Required **release notes** added for new/modified functionality
* [x] NO REQUIRED **updates to other repos** (explain why not required)
# Related Issues
fixes NRLMMD-GEOIPS/geoips#468

# Testing Instructions
* No exhaustive testing required beyond existing and unit tests, as functionality hasn't changed that would impact these

# Summary
* Fixed an issue with log_with_emphasis() function that poorly wrapped file paths over 74 characters in length.
* Used existing wrap() function's functionality for breaking long filenames.
* Added/updated unit tests
* Updated release notes to reflect the fix.

### CODE REVIEW

**Review by evrose54 — CHANGES_REQUESTED** [review]
I generally tend to never use keywords that are reserved in the python namespace. So, for ``geoips/tests/unit_tests/commandline/test_log_setup:insert_word_like_spaces_into_string``, I'd replace the ``str`` argument with ``string`` or ``message`` or something along those lines. Not sure if other people agree, but this is generally something I stick to strictly. Other than that, this PR looks good to me.

**Comment by biosafetylvl5** [comment]
Fixed the `str` conflict! Thanks @evrose54

**Review by evrose54 — APPROVED** [review]
New updates to this branch (replaced ``str`` argument with ``string``) look good to me. Happy to have this merged in!

File: tests/unit_tests/commandline/test_log_setup.py [hunk 1]
```diff
@@ -23,23 +22,53 @@ def generate_random_string(length):
...
+        string = string[:loc] + " " + string[loc + 1 :]
+        loc += random.randint(2, 8)
+    return string
+
+
 def generate_random_messages():
     """Generate a random amount of messages with random length."""
-    num_messages = random.randint(1, 50)
-    return [generate_random_string(random.randint(5, 110)) for _ in range(num_messages)]
+    num_messages = 20
```
Comment (jsolbrig) [inline tests/unit_tests/commandline/test_log_setup.py:51]: Thank you for making this a fixed number of tests. That will fix a lot of confusion.

Is there any reason that 20 wouldn't be a sufficient number of tests?

File: tests/unit_tests/commandline/test_log_setup.py [hunk 2]
```diff
@@ -23,23 +22,53 @@ def generate_random_string(length):
...
     caplog.set_level(logging.INFO)
-    max_message_len = min(74, len(message))
-    assert max_message_len <= 80, "Max emphasis in '*' is longer than 80 chars."
     log_with_emphasis(LOG.info, message)
-    assert "*" * (max_message_len + 6) in caplog.text
-    for wmessage in wrap(message, width=74):
-        assert "** " + wmessage in caplog.text
-    assert "*" * (max_message_len + 6) in caplog.text
+    assert (  # top/bottom of box is formmated correctly
+        "*" * 9  # three for boarders, and min of 5 for string length
```
Comment (jsolbrig) [inline tests/unit_tests/commandline/test_log_setup.py:67]: borders, not boarders

Comment (jsolbrig) [inline tests/unit_tests/commandline/test_log_setup.py:67]: I appreciate the simplified test here. 

Feel free to explain why this isn't needed. Maybe it should test the length of all of the lines to ensure that they are equal unless there is no space in the line? Something like this?

```
stripped = [line.strip('**').strip() for line in lines]
stripped_lens = [len(line) for line in stripped if " " not in line]
assert len(set(stripped_lens)) == 1
```

File: tests/unit_tests/commandline/test_log_setup.py [hunk 3]
```diff
@@ -23,23 +22,53 @@ def generate_random_string(length):
...
+    )
+    assert "** " in caplog.text
+    assert " **" in caplog.text
```
Comment (jsolbrig) [inline tests/unit_tests/commandline/test_log_setup.py:70]: Maybe use `caplog.text.startswith()` and `caplog.text.endswith()` here?

File: tests/unit_tests/commandline/test_log_setup.py
Comment (jsolbrig) [inline tests/unit_tests/commandline/test_log_setup.py:1]: It looks like this doesn't test long filenames. Should it or is that not necessary?

File: tests/unit_tests/commandline/test_log_setup.py (hunk 3 above)
Comment (biosafetylvl5) [reply tests/unit_tests/commandline/test_log_setup.py:70]: Yeah, unfortunately that's not possible because the logging prefix is in the log text. :/ Theoretically, we could filter the log prefix, but that would need updating any time we change the log format

Comment (biosafetylvl5) [reply tests/unit_tests/commandline/test_log_setup.py]: this IS possible with caplog.messages - added 😄

File: tests/unit_tests/commandline/test_log_setup.py
Comment (biosafetylvl5) [reply tests/unit_tests/commandline/test_log_setup.py:1]: Added!

File: tests/unit_tests/commandline/test_log_setup.py (hunk 2 above)
Comment (biosafetylvl5) [reply tests/unit_tests/commandline/test_log_setup.py]: Thanks 😭  Fixed.

File: tests/unit_tests/commandline/test_log_setup.py (hunk 1 above)
Comment (jsolbrig) [reply tests/unit_tests/commandline/test_log_setup.py:76]: Okay, I think this sounds fine to me. If we start having random failures in this, though, we may want to specifically add a few of the edge cases. Things like specific special characters that we expect could cause problems.

I don't see any reason that we would actually expect to have issues with any edge cases, though.

File: tests/unit_tests/commandline/test_log_setup.py
Comment (jsolbrig) [reply tests/unit_tests/commandline/test_log_setup.py:1]: Great, thanks!

**Comment by biosafetylvl5** [comment]
All relevant unit tests and all base tests passing - merging this now! 🎉
